files/image_config/monit/process_checker

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python
+import argparse
+import sys
+import syslog
+
+import psutil
+import swsssdk
+
+
+def check_process_existence(container_name, process_cmdline):
+    """
+    @summary: Check whether the process in the specified container is running or not and
+              an alerting message will written into syslog if it failed to run.
+    """
+    config_db = swsssdk.ConfigDBConnector()
+    config_db.connect()
+    feature_table = config_db.get_table("FEATURE")
+
+    if container_name in feature_table.keys():
+        # We look into the 'FEATURE' table to verify whether the container is disabled or not.
+        # If the container is diabled, we exit.
+        if ("state" in feature_table[container_name].keys()
+                and feature_table[container_name]["state"] == "disabled"):
+            sys.exit(0)
+        else:
+            # We leveraged the psutil library to help us check whether the process is running or not.
+            # If the process entity is found in process tree and it is also in the 'running' or 'sleeping'
+            # state, then it will be marked as 'running'.
+            is_running = False
+            for process in psutil.process_iter(["cmdline", "status"]):
+                if ((' '.join(process.cmdline())).startswith(process_cmdline) and process.status() in ["running", "sleeping"]):
+                    is_running = True
+                    break
+
+            if not is_running:
+                print("'{}' is not running.".format(process_cmdline))
```

Comment:
This script will not be run from the command line. Suggest removing this print statement.

Reply:
Good catch! Initially at here I used the `syslog.syslog(...)` to print the error message if the process was not running. The error message in the syslog will be like the following one:

`ERR monit[22646]: 'lldp|lldpmgrd' status failed (1) -- no output`
`ERR process_checker: 'python /usr/bin/lldpmgrd' is not running`

If I used the `print(...)` statement, the error message will be:

`ERR monit[22050]: 'lldp|lldpmgrd' status failed (1) -- 'python /usr/bin/lldpmgrd' is not running.`

Can you give me some suggestions which format of error message we should choose please?

Reply:
Oh, so Monit captures this stdout and appends it to its log message?

Reply:
Yes,

Reply:
OK. Maybe add a comment above the print statement declaring that if this script is run by Monit, it will append the line to its log message.

Reply:
Updated.